File: school-management-frontend/src/services/studentService.js
import api from '../api/axios';

const studentService = {
  getAllStudents: () => api.get('/students'),
  getStudentById: (id) => api.get(`/students/${id}`),
  searchStudents: (name) => api.get(`/students/search?name=${encodeURIComponent(name)}`),
  getStudentsByClass: (classId) => api.get(`/students/class/${classId}`),
  createStudent: (studentData) => api.post('/students', studentData),
  updateStudent: (id, studentData) => api.put(`/students/${id}`, studentData),
  deleteStudent: (id) => api.delete(`/students/${id}`),
};

export default studentService;

File: school-management-backend/src/main/java/com/school/management/service/StudentService.java
package com.school.management.service;

import com.school.management.dto.StudentRequest;
import com.school.management.entity.Student;

import java.util.List;

public interface StudentService {
    List<Student> getAllStudents();
    Student getStudentById(Long id);
    Student createStudent(StudentRequest request);
    Student updateStudent(Long id, StudentRequest request);
    void deleteStudent(Long id);
    List<Student> searchStudentsByName(String name);
    List<Student> getStudentsByClassId(Long classId);
}
